### PR TITLE
Support hot reload with a prebuilt loader

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Support hot reload with a prebuilt loader.
- [x] Fix unit test stack traces by using **rethrow** instead of **throw error**

@cbracken @tvolkert

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -307,7 +307,7 @@ class AndroidDevice extends Device {
     }
 
     List<String> cmd;
-
+
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/android/android_device.dart]: nit: remove trailing ws

**Review by cbracken — APPROVED** [review]
lgtm. Thanks!

**Review by tvolkert — APPROVED** [review]

**Comment by tvolkert** [comment]
Travis isn't happy :-/

**Comment by johnmccutchan** [comment]
Travis should be happy soon.

**Pull request merged**